analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/Binary.cs

```diff
@@ -219,6 +215,16 @@ private static SymbolicConstraint BinaryConstraint(BinaryOperatorKind kind, Symb
         }
     }
 
+    private static BoolConstraint BinaryNullableBoolConstraint(BinaryOperatorKind kind, BoolConstraint boolConstraint, bool oneIsNull) =>
```

Comment:
The `bool` doesn't help here. More the opposite.
```suggestion
    private static BoolConstraint BinaryNullableBoolConstraint(BinaryOperatorKind kind, BoolConstraint constraint, bool oneIsNull) =>
```